js/pokemon.js

```diff
@@ -1083,11 +1142,14 @@ function saveState() {
 function loadState() {
     var c;
     
-    c = readCookie('generation');
+    c = JSON.parse(readCookie('generation'));
```

Comment:
I expect this will break badly for users who have already played the game and end up with a string rather than array